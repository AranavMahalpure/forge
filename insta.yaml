test:
  auto_accept_unseen: true
  runner: nextest
