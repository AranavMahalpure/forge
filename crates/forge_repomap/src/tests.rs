#[cfg(test)]
mod git_tests {
    use std::path::PathBuf;
    use crate::{GitWalker};
    use insta;

    #[tokio::test]
    async fn test_git_walker_reads_files() {
        // Get the current crate's root directory
        let root_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
        
        // Try reading files
        let walker = GitWalker{};
        let files = walker.read_files(&root_path).await.unwrap();
        
        // Create a sorted list of file paths for snapshot
        let mut paths: Vec<_> = files.keys().collect();
        paths.sort();
        
        insta::assert_snapshot!(format!("{:#?}", paths));
    }
}

mod repomap_tests {
    use std::path::PathBuf;
    use crate::{RepoMap, SymbolIndex, GitWalker};
    use insta;

    #[tokio::test]
    async fn test_create_repomap_for_self() {
        let root_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
        
        let walker = GitWalker{};
        let files = walker.read_files(&root_path).await.unwrap();
        let mut symbol_index = SymbolIndex::new(&root_path);
        symbol_index.generate_from_files(files).await;
        assert!(!symbol_index.definitions().is_empty(), "No definitions found in the repository");
        let repo_map = RepoMap::new()
            .with_map_tokens(2048); 
        
        let map = repo_map.get_repo_map(&symbol_index).await.unwrap();
        
        assert!(!map.is_empty(), "Generated repository map is empty");
        
        insta::assert_snapshot!(map);
    }
}
