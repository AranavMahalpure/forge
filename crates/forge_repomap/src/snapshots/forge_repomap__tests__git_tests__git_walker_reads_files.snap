---
source: crates/forge_repomap/src/tests.rs
expression: "format!(\"{:#?}\", paths)"
---
[
    "Cargo.toml",
    "README.md",
    "src/analyser.rs",
    "src/error.rs",
    "src/file/errors.rs",
    "src/file/git.rs",
    "src/file/mod.rs",
    "src/graph.rs",
    "src/helpers.rs",
    "src/lang/README.md",
    "src/lang/tree-sitter-c-tags.scm",
    "src/lang/tree-sitter-c_sharp-tags.scm",
    "src/lang/tree-sitter-cpp-tags.scm",
    "src/lang/tree-sitter-dart-tags.scm",
    "src/lang/tree-sitter-elisp-tags.scm",
    "src/lang/tree-sitter-elixir-tags.scm",
    "src/lang/tree-sitter-elm-tags.scm",
    "src/lang/tree-sitter-go-tags.scm",
    "src/lang/tree-sitter-java-tags.scm",
    "src/lang/tree-sitter-javascript-tags.scm",
    "src/lang/tree-sitter-kotlin-tags.scm",
    "src/lang/tree-sitter-ocaml-tags.scm",
    "src/lang/tree-sitter-php-tags.scm",
    "src/lang/tree-sitter-python-tags.scm",
    "src/lang/tree-sitter-ql-tags.scm",
    "src/lang/tree-sitter-ruby-tags.scm",
    "src/lang/tree-sitter-rust-tags.scm",
    "src/lang/tree-sitter-typescript-tags.scm",
    "src/lib.rs",
    "src/snapshots/forge_repomap__tests__git_tests__git_walker_reads_files.snap",
    "src/snapshots/forge_repomap__tests__repomap_tests__create_repomap_for_self.snap",
    "src/snapshots/forge_repomap__tests__tests__git_walker_reads_files.snap",
    "src/symbol.rs",
    "src/tests.rs",
    "src/tree_context.rs",
    "src/tree_walker.rs",
    "src/types.rs",
]
